pkg/yamlpath/lexer.go

```diff
@@ -607,6 +644,23 @@ func lexSubPath(l *lexer) stateFn {
 	}
 }
 
+func lexOptionalSuffix(l *lexer) stateFn {
+	if l.peeked(propertyName) {
+		return lexOptionalPropertyName(l)
+	}
+	return lexOptionalArrayIndex(l)
+}
+
+func lexOptionalPropertyName(l *lexer) stateFn {
+	if l.consumed(propertyName) {
+		if l.next() != eof {
+			return l.errorf("property name operator %q may only be used at end of path", propertyName)
```

Comment:
This error is missing a test.

Reply:
Also the error message is inconsistent with other similar error messages in this PR since it includes `%q`. Is this intentional?

Reply:
fixed